auslib/db.py

```diff
@@ -2265,7 +2265,7 @@ def getAllUsers(self, transaction=None):
     def getAllPermissions(self, transaction=None):
         ret = defaultdict(dict)
         for r in self.select(transaction=transaction):
-            ret[r["username"]][r["permission"]] = r["options"]
+            ret[r["username"]][r["permission"]][r["role"]] = r["options"]
```

Comment:
This doesn't look right. Roles are not dependent on a Permission, just on the User.